### PR TITLE
CDN deployment conditional: Bring back `BUILD_VCS_NUMBER`

### PR DESCRIPTION
**Pull request by Schniz**
### 🔦 Summary
Removing `BUILD_VCS_NUMBER` in #2059 broke Perfer CI in Thunderbolt.
Let's return it and discuss afterwards whether we want to add another environment variable to Perfer CI:

* Adding a fake `ARTIFACT_VERSION` env var can do strange things to some projects' build pipelines
* Adding a fake `SRC_MD5` is very weird

### 🗺️ Test Plan
After releasing a new version, Perfer builds will pass again.

### CODE REVIEW

**Comment by Schniz** [comment]
cc @NitayRabi is there a problem with that?

**Comment by ronami** [comment]
cc @ranyitz  @netanelgilad

**Comment by netanelgilad** [comment]
It's true that not all CI builds have SRC_MD5 and we need to fallback to BUOLD_VCS_NUMBER but that should be the order, SRC_MD5 should take precedence.

File: packages/yoshi-helpers/src/utils.ts [hunk 1]
```diff
@@ -193,8 +193,8 @@ export const getProjectArtifactVersion = () => {
   return (process.env.ARTIFACT_VERSION
     ? // Dev CI
       process.env.ARTIFACT_VERSION.replace('-SNAPSHOT', '')
-    : // PR CI won't have a version, only SRC_MD5
-      process.env.SRC_MD5) as string;
+    : // PR CI won't have a version, only VCS_BUILD_NUMBER or SRC_MD5
+      process.env.SRC_MD5 || process.env.VCS_BUILD_NUMBER) as string;
```
Comment (Schniz) [inline packages/yoshi-helpers/src/utils.ts]: @netanelgilad SRC_MD5 is before

Comment (netanelgilad) [reply packages/yoshi-helpers/src/utils.ts]: yep github mobile got me confused :D

**Comment by Schniz** [comment]
We're adding SRC_MD5 in Perfer CI

**Pull request closed without merging**